### PR TITLE
Zeus - Fix remote control breaking Racks and resetting custom PTTs

### LINKED ISSUE #1267: Zeus - Exiting remote control while the AI character is inside a vehicle breaks racks and intercom
**Mods**
- **Arma 3:** `2.12.150779`
- **CBA:** `3.15.8.220912`
- **ACRE2:** `2.11.0.1054`
- **ACE3:** `3.15.2`

**Description:**
Zeus remote control - exiting remote control while the AI character is inside a vehicle causes vehicle rack radios and intercom radios to be bugged for the player character and other remotely controlled AI characters.

**Steps to reproduce:**
1. Spawn a vehicle (I tested both Eden editor and Zeus spawning). I spawned a HEMTT truck just to show that it happens, even if the vehicle used to "activate" the bug doesn't need to have a radio nor intercom.
2. Remote control an AI character (both inside a vehicle or spawned outside).
3. Exit remote control while the AI character you control is inside a vehicle.
4. Enter another vehicle (one with a rack radio or intercom radio) as a player character or another AI character.
5. Try to use rack radio (either select the "use" option in the ACE rack radio menu or select work / monitor A on the intercom panel).
6. On the rack radio / intercom UI, the "DASH R/T" text will briefly turn white...
7. ... and then, in the rack radio scenario: the rack radio UI will disappear and reappear after a second. "DASH" text will be gray, and you won't be able to use the radio from the rack;
In the intercom scenario: intercom text UI will disappear and reappear after you close the intercom panel UI. "CREW R/T" text will be white, but "DASH" text will be gray, and you won't be able to use the radio from the intercom / rack.
8. In the brief time while the text is white, you can use the racked radio, but that ability will end the second the text stops being white.

**Expected behavior:**
Exiting remote control of an AI character inside a vehicle should not render rack radios in other vehicles unusable for the player character and other remotely controlled AI characters.

**Workaround:**
- Exit the remote control while the AI character is outside the vehicle to avoid the bug in the first place.

**How to temporarily fix the bug:**
- Enter Zeus mode. Select the AI character with which you activated the bug and put him outside the vehicle. (Killing him inside or destroying / despawning the vehicle won't work and will make the bug unfixable.) You can then put him back inside by clicking and dragging him back into the vehicle.
- Or remote control the same AI character you activated the bug with, exit the vehicle, and leave the remote control.
- Upon again doing the steps to reproduce, the issue will reappear and will work identically.

**Where did the issue occur?**
- Self-Hosted Multiplayer

**Log Files:**
- https://pastebin.com/9WuGY0cu
- https://pastebin.com/H3q7v9P1 [I only tested this bug for like 3 last days]

**Additional context:**
- **I can confirm this bug is not only on my end because I asked my friend to reproduce this bug, and it behaved as described here.**
- The bug does not occur after exiting the mission and restarting it, but after doing the same thing, the bug will return.
- Occurrence chance for me (and my friend) when following the aforementioned instructions: 100%
- I didn't have a chance to test if this bug affects anything else besides not being able to use rack radios / intercom radios in vehicles (like, for example, actual intercom communication with other players)

**What I tested and didn't work:**
- Disconnecting from the intercom before exiting the remote control without leaving the vehicle.
- Switching the radio off before exiting the remote control without leaving the vehicle.
- Setting different options in the intercom panel UI (setting work or monitor to any options, push-to-talk, v/o, volume) before exiting the remote control without leaving the vehicle.
- Dying and respawning after the bug was activated
- Deleting unit and vehicle I used to activate the bug
- Killing the unit or destroying the vehicle I used to activate the bug (destroying the vehicle kind of worked a couple of times because the AI character exited the vehicle the moment it blew up)
- Using units and vehicles spawned before and after activating the bug.
- Restarting TS3 plugin
- Turning off TS3 and starting it again

### PR DESCRIPTION
**Pull request by mrschick**
**When merged this pull request will:**
- Fix Racks being broken for the Zeus Player when remote controlling AI in vehicles, which resolves #1267:
It was caused by the `sys_rack/fnc_rackPFH.sqf` PFH being started when entering that AI and not terminated when exiting it, resulting in the PFH constantly disconnecting the player from his Racks. My fix works by limiting Rack initialization and PFH launch to only player units entering a vehicle, effectively preventing use of Racks by RCd AI entirely.
- Fix custom PTT assignments being reset when using a remote controlled unit's radios:
A minor bug I encountered some time ago, happened because there was only a single `DVAR(ACRE_ASSIGNED_PTT_RADIOS)` that stored custom PTT assignments, when controlling a unit and using its radios `sys_core/fnc_handleMultiPttKeyPress.sqf` would check for the custom PTT radios to be present, not find them on the AI and reset the list. Now there are 2 DVARs to remember custom PTT assignments on the player and last controlled AI unit.

### CODE REVIEW

**Review by Mike-MF — APPROVED** [review]
Works

File: addons/sys_core/XEH_preInit.sqf [hunk 1]
```diff
@@ -66,6 +66,7 @@ DVAR(ACRE_LISTENER_DIVE) = 0;
 
 DVAR(ACRE_PTT_RELEASE_DELAY) = 0.2;
 DVAR(ACRE_ASSIGNED_PTT_RADIOS) = [];
+DVAR(ACRE_ASSIGNED_PTT_RADIOS_REMOTE) = [];
```
Comment (jonpas) [inline addons/sys_core/XEH_preInit.sqf]:
```suggestion
DVAR(ACRE_ASSIGNED_PTT_RADIOS_REMOTE) = []; // Remote controlled unit
```

**Review by jonpas — COMMENTED** [review]
How does this work when controlling a UAV, as that is also remote control?

> My fix works by limiting Rack initialization and PFH launch to only player units entering a vehicle, effectively preventing use of Racks by RCd AI entirely.

It's more of a workaround but it does prevent an issue, so I am fine with accepting this change, but we should add a note in documentation.

**Comment by jonpas** [comment]
@BrettMayson Since you authored all of the Zeus compatibility, mind taking a look at this?

**Comment by mrschick** [comment]
> How does this work when controlling a UAV, as that is also remote control?

I did not think of testing that. Unless it also changes `acre_player` from `player`, it should not be affected.
Should get around to testing it next week.

> It's more of a workaround but it does prevent an issue, so I am fine with accepting this change, but we should add a note in documentation.

Maybe we can figure out an actual fix. Didn't look too much into it due to not seeing a use case in which a RCd AI should have to use Racks.

**Comment by mrschick** [comment]
> How does this work when controlling a UAV, as that is also remote control?

Turns out `addons/sys_rack/fnc_enterVehicle.sqf` is not called when controlling a drone, only when entering a vehicle, so I can't find any cases in which it would create issues.
Unless a UAV has Racks, though it doesn't seem that is supported ATM.

**Comment by mrschick** [comment]
Haven't looked too deep into it, but given the lack of apparent use-cases I don't think implementing Racks being accessible by RCd AI is worth the time.

**Review by BrettMayson — APPROVED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**